src/plugins/data_source_management/public/components/edit_data_source/components/header/header.tsx

```diff
@@ -25,22 +26,51 @@ export const Header = ({
   isFormValid,
   onClickDeleteIcon,
   onClickTestConnection,
+  onClickSetDefault,
   dataSourceName,
+  isDefault,
 }: {
   showDeleteIcon: boolean;
   isFormValid: boolean;
   onClickDeleteIcon: () => void;
   onClickTestConnection: () => void;
+  onClickSetDefault: () => void;
   dataSourceName: string;
+  isDefault: boolean;
 }) => {
   /* State Variables */
   const [isDeleteModalVisible, setIsDeleteModalVisible] = useState(false);
+  const [isDefaultDataSourceState, setIsDefaultDataSourceState] = useState(isDefault);
 
   const changeTitle = useOpenSearchDashboards<DataSourceManagementContext>().services.chrome
     .docTitle.change;
 
   changeTitle(dataSourceName);
 
+  const setDefaultAriaLabel = i18n.translate(
+    'dataSourcesManagement.editDataSource.setDefaultDataSource',
+    {
+      defaultMessage: 'Set as default Data Source.',
+    }
+  );
+
+  const renderDefaultIcon = () => {
+    return (
+      <EuiButtonEmpty
+        onClick={() => {
+          onClickSetDefault();
+          setIsDefaultDataSourceState(!isDefaultDataSourceState);
+        }}
+        disabled={isDefaultDataSourceState}
+        iconType={isDefaultDataSourceState ? 'starFilled' : 'starEmpty'}
+        aria-label={setDefaultAriaLabel}
+        data-test-subj="datasource-edit-setDefaultDataSource"
+      >
+        {isDefaultDataSourceState ? 'Default' : 'Set as default'}
+      </EuiButtonEmpty>
+    );
+  };
```

Comment:
Also should we make UI changes similar to index patterns screen? 
<img width="1914" alt="Screenshot 2024-03-18 at 2 39 07 PM" src="https://github.com/opensearch-project/OpenSearch-Dashboards/assets/63824432/6f02c607-02ac-4bbb-9684-544e22c4fe62">

So on index pattern, there is just star and on tool tip we display a message `Set as default index.`. @kgcreative any thoughts?

Reply:
Hi Bandini. THanks for pointing this out. I have checked with Xenia on this. Let me find the Figma link and add it here!